File: YFDebugPannel/YFDebugPannel/HCSwitchCell.m
#import "HCSwitchCell.h"
#import "HCCellItem.h"

@interface HCSwitchCell ()

@property (nonatomic, strong) UISwitch *toggle;

@end

@implementation HCSwitchCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    if (self = [super initWithStyle:UITableViewCellStyleDefault reuseIdentifier:reuseIdentifier]) {
        _toggle = [[UISwitch alloc] init];
        [_toggle addTarget:self action:@selector(switchChanged:) forControlEvents:UIControlEventValueChanged];
        self.accessoryView = _toggle;
        self.selectionStyle = UITableViewCellSelectionStyleNone;
    }
    return self;
}

- (void)configureWithItem:(HCCellItem *)item {
    self.textLabel.text = item.title;
    self.toggle.on = [item.value boolValue];
    self.toggle.enabled = item.enabled;
}

- (void)switchChanged:(UISwitch *)sender {
    if (self.valueChanged) {
        self.valueChanged(sender.isOn);
    }
}

@end

File: YFDebugPannel/YFDebugPannel/HCSwitchCell.swift
import UIKit

final class HCSwitchCell: UITableViewCell {
    var valueChanged: ((Bool) -> Void)?

    private let toggle = UISwitch()

    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: .default, reuseIdentifier: reuseIdentifier)
        toggle.addTarget(self, action: #selector(switchChanged(_:)), for: .valueChanged)
        accessoryView = toggle
        selectionStyle = .none
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }

    func configure(with item: HCCellItem) {
        textLabel?.text = item.title
        toggle.isOn = boolValue(item.value)
        toggle.isEnabled = item.enabled
    }

    @objc private func switchChanged(_ sender: UISwitch) {
        valueChanged?(sender.isOn)
    }
}
